lib/ddtrace/utils.rb

```diff
@@ -1,9 +1,21 @@
+require 'thread'
+
 module Datadog
-  # TODO[manu]: write docs
+  # Utils contains low-level utilities, typically to provide pseudo-random trace IDs.
   module Utils
+    # We use a custom random number generator because we want no interference
+    # with the default one. Using the default prng, we could break code that
+    # would rely on srand/rand sequences.
+    @rnd = Random.new
```

Comment:
@palazzem our sequence of random numbers won't be reproducible (even, per thread) as all threads will use the same generator. But we don't care, all we need is no collision, our code does not rely on peculiar sequences of IDs, the more "pseudo-random", the better. But we definitely need to *not* interfere with the sequence of the main program.